code/datums/diseases/advance/symptoms/heal.dm

```diff
@@ -72,13 +76,35 @@
 	if(A.totalStageSpeed() >= 6)
 		power = 2
 
+/datum/symptom/heal/starlight/proc/CanTileHeal(turf/T)
+	var/area/area_to_check = get_area(T)
+	if(isspaceturf(T) || area_to_check.outdoors) // Outdoors covers lavaland and unroofed areas but with tiles under, while space covers normal space and those caused by explosions
+		return STARLIGHT_CAN_HEAL
+	else if(istype(T, /turf/open/openspace)) // is the turf openspace
+		T = T.below()
+		area_to_check = get_area(T)
+		if(isspaceturf(T) || area_to_check.outdoors)
+			return STARLIGHT_CAN_HEAL
+	else if(istransparentturf(T)) // is the turf glass/transparent
+		T = T.below()
+		if(!T) // no below turf exists, assume its space
+			return STARLIGHT_CAN_HEAL_WITH_PENALTY
+		area_to_check = get_area(T)
+		if(isspaceturf(T) || area_to_check.outdoors)
+			return STARLIGHT_CAN_HEAL_WITH_PENALTY // space tile is underneath glass, so near space
+	return STARLIGHT_CANNOT_HEAL
```

Comment:
Does this account to for the tile above you? could you do something like calling CanTileHeal on above()? or perhaps just the openspace/transparent stuff?

it'd be nice if this could be done recursively, or maybe with a while loop? yeah use a while loop just in case some asshole makes a 100z map someday

Reply:
The tile above should never make a difference I think, since if there's a space turf above you (and not a roof in the way) your current area has to be space/outdoors to begin with to have no roof.
The while loop thing was added, and just in case of 100z the two glass floors limitation should prevent that mess

Reply:
If you're under a glass tile that's on the top z level of a station, should you get the benefits of being in starlight?

Reply:
After some testing yes that should be the case